skytap/published_service.go

```diff
@@ -172,10 +177,48 @@ func (s *PublishedServicesServiceClient) Delete(ctx context.Context, environment
 		return err
 	}
 
-	_, err = s.client.do(ctx, req, nil)
+	_, err = s.client.do(ctx, req, nil, nil, nil)
 	if err != nil {
 		return err
 	}
 
 	return nil
 }
+
+func (payload *CreatePublishedServiceRequest) compare(ctx context.Context, c *Client, v interface{}, state *environmentVMRunState) (string, bool) {
+	if serviceOriginal, ok := v.(*PublishedService); ok {
+		publishedService, err := c.PublishedServices.Get(ctx, *state.environmentID, *state.vmID, *state.adapterID, *serviceOriginal.ID)
+		if err != nil {
+			return requestNotAsExpected, false
+		}
+		actual := CreatePublishedServiceRequest{
+			publishedService.InternalPort,
+		}
+		if payload.string() == actual.string() {
+			return "", true
+		}
+		return "published service not ready", false
+	}
+	log.Printf("[ERROR] SDK published service comparison not possible on (%v)\n", v)
+	return requestNotAsExpected, false
+}
+
+func (payload *UpdatePublishedServiceRequest) compare(ctx context.Context, c *Client, v interface{}, state *environmentVMRunState) (string, bool) {
+	return payload.CreatePublishedServiceRequest.compare(ctx, c, v, state)
```

Comment:
I'm unclear on why we have an `Update` function and an `UpdatePublishedServiceRequest` type. The API only supports create and destroy operations, not update. Do we still need the Update function?

Reply:
Point taken - the function is here to support the terraform provider which requires an 'update' step. Since this is actually a delete and create I will rename the function and corresponding data structures accordingly.